src/stores/app.store.ts

```diff
@@ -0,0 +1,394 @@
+  }
+  auth: {
+    social: SocialAuthState[]
+    passwordless: {
+      javascriptOrigin: string
+      redirectUri: string
+    }
+    wallet: {
+      websiteDomain: string
+      selectedTheme: Theme
+      hasUIMode: boolean
+      hasUIModeInGateway: boolean
+    }
+    redirectUri: string
+  }
+  store: {
+    userLimits: {
+      storage: UserLimitState
+      bandwidth: UserLimitState
+    }
+    region: StorageRegion
+  }
+}
+
+function calculateUserLimits(userLimit: number): UserLimitState {
+  if (userLimit < MAX_DATA_TRANSFER_BYTES) {
+    const isUnder1GB = userLimit < bytes('1 GB')
+
+    const calculatedUserLimit = bytes(userLimit, {
+      unitSeparator: ' ',
+      unit: isUnder1GB ? 'MB' : 'GB',
+    })
+
+    const [value, unit] = calculatedUserLimit.split(' ') as [
+      number,
+      UserLimitUnit
+    ]
+    return { value, unit, isUnlimited: false }
+  } else {
+    return { isUnlimited: true }
+  }
+}
+
+export const useAppStore = defineStore('app', {
+  state: (): AppState => ({
+    appName: '',
+    appId: '',
+    appAddress: '',
+    access: {
+      selectedChain: 'none',
+    },
+    logos: {
+      dark: {
+        horizontal: '',
+        vertical: '',
+      },
+      light: {
+        horizontal: '',
+        vertical: '',
+      },
+    },
+    auth: {
+      social: [],
+      passwordless: {
+        javascriptOrigin: '',
+        redirectUri: '',
+      },
+      redirectUri: '',
+      wallet: {
+        websiteDomain: '',
+        selectedTheme: 'dark',
+        hasUIMode: false,
+        hasUIModeInGateway: false,
+      },
+    },
+    store: {
+      userLimits: {
+        storage: {
+          isUnlimited: true,
+        },
+        bandwidth: {
+          isUnlimited: true,
+        },
+      },
+      region: 'asia',
+    },
+  }),
+  getters: {
+    appConfigRequestBody(state): AppConfig {
+      let storage_limit: number, bandwidth_limit: number
+
+      const storageLimit = state.store.userLimits.storage
+      const bandwidthLimit = state.store.userLimits.bandwidth
+      if (storageLimit.isUnlimited) {
+        storage_limit = MAX_DATA_TRANSFER_BYTES
+      } else {
+        storage_limit = bytes(`${storageLimit.value} ${storageLimit.unit}`)
+      }
+
+      if (bandwidthLimit.isUnlimited) {
+        bandwidth_limit = MAX_DATA_TRANSFER_BYTES
+      } else {
+        bandwidth_limit = bytes(
+          `${bandwidthLimit.value} ${bandwidthLimit.unit}`
+        )
+      }
+
+      const socialAuth = state.auth.social
+      const cred: {
+        verifier: string
+        clientId?: string
+        clientSecret?: string
+        redirectURL?: string
+        origin?: string
+      }[] = socialAuth.map((authType) => {
+        return {
+          verifier: authType.verifier,
+          clientId: authType.clientId,
+          clientSecret: authType.clientSecret,
+          redirectURL: authType.redirectUri,
+        }
+      })
+
+      if (
+        state.auth.passwordless.javascriptOrigin &&
```

Comment:
Add this getter as a function to gateway service file

Reply:
Resolved at https://github.com/arcana-network/developer-dashboard/pull/119/commits/4a31142c17d9c5f2f9f006a070b7df8416eb0686